framework/rbac/DbManager.php

```diff
@@ -524,7 +524,7 @@ public function getPermissionsByRole($roleName)
      */
     public function getPermissionsByUser($userId)
     {
-        if (empty($userId)) {
+        if (!is_numeric($userId) && empty($userId)) {
```

Comment:
The condition is:

> When userId is NOT numeric and is empty - return empty array.

I don't think this change will cause problems with non-integer IDs, but @dmirogin, just to make sure that the change is OK, could add a test against strings and Value Objects like this one:

```
class UserID
{
    private $id;

    public function __construct($id)
    {
        $this->id = $id;
    }

    public function __toString()
    {
        return $this->id;
    }
}
````